src/main/java/com/google/sps/servlets/DeleteRecipeServlet.java

```diff
@@ -16,11 +25,36 @@
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    long id = Long.parseLong(request.getParameter("id"));
+    long privateRecipeID = Long.parseLong(request.getParameter("id"));
+    String name = request.getParameter("name");
+    String description = request.getParameter("description");
 
-    // PrivateRecipes are the user uploaded recipes so only PrivateRecipes can be deleted (right now)
-    Key recipeEntitykey = KeyFactory.createKey("PrivateRecipe", id);
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-    datastore.delete(recipeEntitykey);
+
+    /// delete the private recipe
+    Key privateKey = KeyFactory.createKey("PrivateRecipe", privateRecipeID);
+    datastore.delete(privateKey);
+
+    // make filters to find the public recipe
+    Filter nameFilter = new FilterPredicate("name", FilterOperator.EQUAL, name);
+    Filter descriptionFilter = new FilterPredicate("description", FilterOperator.EQUAL, description);
+    // TODO: further filter by author ID
+    Filter composFilter = CompositeFilterOperator.and(nameFilter, descriptionFilter);
+
+    Query query = new Query("PublicRecipe").setFilter(composFilter);
+    PreparedQuery results = datastore.prepare(query);
+    
+    try {
+      Entity publicRecipeEntity = results.asSingleEntity();
+      // publicRecipeEntity will only equal null if there is no entity left after the filters are applied
+      if (publicRecipeEntity != null) {
+        long publicRecipeID = publicRecipeEntity.getKey().getId();
+        Key recipeEntitykey = KeyFactory.createKey("PublicRecipe", publicRecipeID);
+        datastore.delete(recipeEntitykey);
+      }
+    } catch (TooManyResultsException e) {
```

Comment:
So while this is correct error to catch, simply not doing the transaction of deleting the public recipe doesn't seem like the correct result. If for some reason this error were to ever be caught, there would no longer be a private recipe to reference, but there are more than one public recipes with the same name, author, and recipe still in Datastore; this is fatal because the user expects one of them to not exist anymore. I would suggest looking at a different way of approaching the solution, and asking Alex or Olivia for more feedback about whether the areas where this implementation could struggle are worth accounting.

Reply:
yeah no I'll just do what you suggested of storing the public recipe ID in the private recipe